app/web/src/utils/typescriptLinter.ts

```diff
@@ -120,12 +120,25 @@ export const createTypescriptSource = async (
     const doc = view.state.doc;
     // We could be more efficient by updating only the changed spans
     const docString = doc.toString();
+
+    let diagnostics: Diagnostic[] = [];
+
+    // custom lint rule to ensure that we have a `main` function entrypoint
+    if (!docString.includes("function main(")) {
+      diagnostics = diagnostics.concat(
+        diagnosticsForMessage(
+          1,
+          1,
+          "Function should include a `main` function for code execution",
```

Comment:
This may be confusing if the user wrote `const main = async () => {}`, which won't execute properly, but the message suggest that's possible.

Reply:
I do love how simple this is!

Reply:
Ok, @paulocsanz and I talked in slack and we decided this is fine for now but we can fine tune if it becomes a problem later